plugin/daemon_example/daemon_example.cc

```diff
@@ -175,11 +175,9 @@ static int daemon_example_plugin_deinit(void *p __attribute__ ((unused)))
   my_write(con->heartbeat_file, (uchar*) buffer, strlen(buffer), MYF(0));
 
   /*
-    Need to wait for the hearbeat thread to terminate before closing
-    the file it writes to and freeing the memory it uses.
+    As thread is joined, we can close the file it writes to and
+    freeing the memory it uses.
```

Comment:
"We can freeing" does not compute. I’d move this comment before the `my_write()` call, because also that call could race with `mysql_heartbeat()`.